app/main.py

```diff
@@ -1,6 +1,17 @@
-from typing import Callable
+from typing import Callable, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    data = {}
+
+    def inner(*args) -> Any:
+        if args not in data:
+            print("Calculating new result")
+            result = func(*args)
+            data[args] = result
+            return result
+        if args in data:
```

Comment:
The check `if args in data:` is redundant because if the code reaches this line, it means the condition `if args not in data:` was false, implying `args` is already in `data`. You can remove this line and the associated block to simplify the code.